htdocs/core/lib/pdf.lib.php

```diff
@@ -1410,29 +1410,68 @@ function pdf_getlinedesc($object, $i, $outputlangs, $hideref = 0, $hidedesc = 0,
 {
 	global $db, $conf, $langs;
 
-	$idprod = (!empty($object->lines[$i]->fk_product) ? $object->lines[$i]->fk_product : false);
-	$label = (!empty($object->lines[$i]->label) ? $object->lines[$i]->label : (!empty($object->lines[$i]->product_label) ? $object->lines[$i]->product_label : ''));
-	$desc = (!empty($object->lines[$i]->desc) ? $object->lines[$i]->desc : (!empty($object->lines[$i]->description) ? $object->lines[$i]->description : ''));
-	$ref_supplier = (!empty($object->lines[$i]->ref_supplier) ? $object->lines[$i]->ref_supplier : (!empty($object->lines[$i]->ref_fourn) ? $object->lines[$i]->ref_fourn : '')); // TODO Not yet saved for supplier invoices, only supplier orders
-	$note = (!empty($object->lines[$i]->note) ? $object->lines[$i]->note : '');
-	$dbatch = (!empty($object->lines[$i]->detail_batch) ? $object->lines[$i]->detail_batch : false);
+	$multilangsactive = getDolGlobalInt('MAIN_MULTILANGS');
 
-	if ($issupplierline) {
-		include_once DOL_DOCUMENT_ROOT.'/fourn/class/fournisseur.product.class.php';
-		$prodser = new ProductFournisseur($db);
-	} else {
-		include_once DOL_DOCUMENT_ROOT.'/product/class/product.class.php';
-		$prodser = new Product($db);
+    if ($issupplierline) {
+        include_once DOL_DOCUMENT_ROOT.'/fourn/class/fournisseur.product.class.php';
+        $prodser = new ProductFournisseur($db);
+    } else {
+        include_once DOL_DOCUMENT_ROOT.'/product/class/product.class.php';
+        $prodser = new Product($db);
 
-		if (!empty($conf->global->PRODUIT_CUSTOMER_PRICES)) {
-			include_once DOL_DOCUMENT_ROOT . '/product/class/productcustomerprice.class.php';
+        if (!empty($conf->global->PRODUIT_CUSTOMER_PRICES)) {
+            include_once DOL_DOCUMENT_ROOT . '/product/class/productcustomerprice.class.php';
+        }
+    }
+
+	//id
+	$idprod = (!empty($object->lines[$i]->fk_product) ? $object->lines[$i]->fk_product : false);
+    if($idprod) {
+        $prodser->fetch($idprod);
+        //load multilangs
+        if($multilangsactive){
+            $prodser->getMultiLangs();
+            $object->lines[$i]->multilangs = $prodser->multilangs;
+        }
+    }
+	//label
+	if (!empty($object->lines[$i]->label)) {
+		$label = $object->lines[$i]->label;
+	} else {
+		if (!empty($object->lines[$i]->multilangs[$langs->defaultlang]['label']) && $multilangsactive) {
```

Comment:
For PDF,what we want to use is the language of recipient of the pdf, so $outputlangs, not $langs. Am i wrong ?

Reply:
You're right : fixed !